### DISCUSSION TITLE
Static RBD PV mount options

### DISCUSSION THREAD

**Discussion by LukeMauldin** [ceph/ceph-csi#3307] [General]
# Describe the bug #

Following documentation: https://github.com/ceph/ceph-csi/blob/devel/docs/static-pvc.md#create-rbd-static-pv
There is no documented way to specify mount options such as `noatime,discard`.  When using ceph with storageclass to create dynamic PVC, can specify mount options: https://github.com/ceph/ceph-csi/blob/devel/examples/rbd/storageclass.yaml#L147-L148

Need a way to specify mount options when mounting rbd static pv.

# Environment details #

- Image/version of Ceph CSI driver : v3.6.0

**Comment by Madhu-1** [comment]
@LukeMauldin, have you tried creating a storageclass and adding the storageclass name in the pvc/PV object?

**Comment by LukeMauldin** [comment]
If I use a storageclass I am able to specify the mount options but I am following the static-pv example where the Ceph rbd volume is already created and there is no storageclass specified in that example.

**Comment by Madhu-1** [comment]
Even I case of static pvc you can specify the storageclass to fetch the mount options. Examples might not be covering everything.

**Comment by LukeMauldin** [comment]
Can you please give me an example?

**Comment by LukeMauldin** [comment]
Is there anyone who can provide an example of using a storageclass with a static pvc?

**Comment by Madhu-1** [comment]
You have `mountOptions` in PV itself you can use it like below?

```yaml
kubectl get pv/pvc-42b935f9-96d3-4c33-be07-f9826cb8ab43 -oyaml
apiVersion: v1
kind: PersistentVolume
metadata:
  annotations:
    pv.kubernetes.io/provisioned-by: rook-ceph.rbd.csi.ceph.com
  creationTimestamp: "2022-08-16T11:05:15Z"
  finalizers:
  - kubernetes.io/pv-protection
  name: pvc-42b935f9-96d3-4c33-be07-f9826cb8ab43
  resourceVersion: "4658"
  uid: babdfa85-e591-4174-a09f-e4d41476b41a
spec:
  accessModes:
  - ReadWriteOnce
  capacity:
    storage: 1Gi
  claimRef:
    apiVersion: v1
    kind: PersistentVolumeClaim
    name: rbd-pvc
    namespace: default
    resourceVersion: "4653"
    uid: 42b935f9-96d3-4c33-be07-f9826cb8ab43
  csi:
    controllerExpandSecretRef:
      name: rook-csi-rbd-provisioner
      namespace: rook-ceph
    driver: rook-ceph.rbd.csi.ceph.com
    fsType: ext4
    nodeStageSecretRef:
      name: rook-csi-rbd-node
      namespace: rook-ceph
    volumeAttributes:
      clusterID: rook-ceph
      imageFeatures: layering
      imageFormat: "2"
      imageName: csi-vol-4eac376d-1d53-11ed-b8c5-daca746cf3c6
      journalPool: replicapool
      pool: replicapool
      storage.kubernetes.io/csiProvisionerIdentity: 1660646054917-8081-rook-ceph.rbd.csi.ceph.com
    volumeHandle: 0001-0009-rook-ceph-0000000000000001-4eac376d-1d53-11ed-b8c5-daca746cf3c6
  mountOptions:
  - discard noatime
  persistentVolumeReclaimPolicy: Delete
  storageClassName: rook-ceph-block
  volumeMode: Filesystem
status:
  phase: Bound
```
